scripts/build_native.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -e
+
+BASEDIR="$(readlink -f $(dirname $0)/..)"
+
+if [ ! -f "Cargo.toml" ]; then
+    >&2 echo "Your current working directory doesn't look like we'll find the sources to build. This script must be run from from a checked out copy of the kanidm/kanidm project root."
+    exit 1
+fi
+
+# Expected target format: debian-12-aarch64-unknown-linux-gnu
+target=${1?}
+
+. /etc/os-release
+
+echo "Building for: ${target} on ${PRETTY_NAME}"
+rustup target add "$target"
```

Comment:
I think you might want to add an explicit `rustup toolchain install` because of [this recent change to rustup](https://github.com/rust-lang/rustup/pull/3985)

Reply:
Funny enough, in CI I think this is actually redundant because we ensure rust target in GHA magic, but a human may run into this so I'll have a tweak, and less magic is better anyway if possible.

Reply:
Moved it into the dep install phase as a `rustup install`. We do still need the GHA magic because we're mucking with sccache & mold, but now it's a wee bit more future proof for human users.

Reply:
Actually, no, can't do it like that because it's a chicken & egg, need deps to install rust, need rust to use rustup. So yes, it's now more future proof, but still needs to live in the build script which is a bit annoying.